### ISSUE TITLE
Add wrapper functionality to consul-template similar to envconsul.

### ISSUE THREAD

**Issue by caquino** [hashicorp/consul-template#439]
Hi,

If we run consul-templates inside of containers, we need to build a script that's the ENTRYPOINT/RUN  running consul-template and also runs the application that you need.

To simplify container operations it would be really usefull if consul-template could do something similar to envconsul, working as a wrapper around the application.

Let's imagine that we want to run nginx logging to stderr/stdout and having the nginx.conf managed by consul-template.

Something similar to this command line could be used:

```
consul-template \
  -consul 127.0.0.1:8500 \
  -template "/tmp/template.ctmpl:/etc/nginx/nginx.conf:/usr/sbin/nginx -c /etc/nginx/nginx.conf" \
  -retry 30s \
  -signal SIGHUP \
  -foreground
````

Basically it would run nginx using the cmdline /usr/sbin/nginx -c /etc/nginx/nginx.conf (which has daemon off; on the config) and send SIGHUPs to the application when a reload is necessary.

This would make consul-template even more simple for container execution.

Thanks

**Comment by sethvargo** [comment]
Hi @caquino 

The recommended way to do this in Consul Template is to use a custom bash script. That script can respond to signals, but Consul Template does not monitor the child process. In fact, the child process must exit in 30s or CT will crash.

You can write a custom script (or use an init system), but Consul Template is not designed to be an init system.
